app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardItemViewHolder.kt

```diff
@@ -25,7 +25,7 @@ class CreditCardItemViewHolder(
     fun bind(creditCard: CreditCard) {
         // TODO this should be last4 instead...
         //  and option to explicitly decrypt if necessary to show full number
-        credit_card_number.text = creditCard.encryptedCardNumber.number
+        credit_card_number.text = creditCard.cardNumberLast4
```

Comment:
I am wondering how come we didn't simply add the ellipses here, but mutated the state instead.

Reply:
My other thinking would be to have a getter to return the obfuscated number so that it can be reusable.

Reply:
🤷 It's a viewHolder so I thought maybe it shouldn't have any logic there but I suppose it could work.
A getter on the `CreditCard` in AC would be preferable indeed, but I tried to fix it all in Fenix.

Reply:
Let's go with adding a getter function something like `getObfuscatedCardNumber` in AC to complete this. You will notice 2 different variations of the `CreditCard` (one in concept-engine to match what we provide to GV which requires the card number to be decrypted, and concept-storage). This should also help us resolve https://github.com/mozilla-mobile/android-components/issues/10248 much more easily